LDK/resources/web/LDK/panel/TabbedReportPanel.js

```diff
@@ -1067,12 +1072,16 @@ Ext4.define('LDK.panel.TabbedReportPanel', {
 
             for (var i=0;i<token.length;i++){
                 var t = token[i].split(':');
+                t[0] = decodeURIComponent(t[0]);
```

Comment:
This is probably desirable; however, is there background or an issue on what drove this?  as you probably know, report names are defined in the database and currently have little restriction over them. historically URL encoding or lack thereof was a problem (also historically there had been bugs around double-encoding). Given how long it's been stable (to my knowledge), I'm surprised changes are required.  was there a specific bug, was it just to type-convert t[1], or is this added because it seemed correct?

Reply:
I think this was a long-standing bug. For the original repro case see here: https://www.labkey.org/ONPRC/Support%20Tickets/issues-details.view?issueId=33875

Reply:
@labkey-jeckels I agree with the change, which effectively moves the decodeURIComponent() to the beginning of the for() loop, rather than within the switch(). I'm probably being dense, but I'm not understanding why the code before didnt work.  At least on my 20.7 enlistment, line 1063 still decoded 'activeReport', which seems like it should have solved the issue from ticker 33875:

                switch(t[0]){
                    case 'inputType':
                        context.inputType = t[1];
                        break;
                    case 'showReport':
                        this.isReportTabSelected = (t[1] === 1);
                        break;
                    case 'activeReport':
                        var report = decodeURI(t[1]);


I understand "this.isReportTabSelected = (t[1] === 1);" was a bug, and I can see an argument for putting the decodeURIComponent() first, but shouldnt the 20.7 code have also fixed that issue?

Reply:
I looked into the history and it looks like we had different fixes for the same problem applied in different branches. I think you're right that the 20.7 variant had already addressed the problem, likely with the fix for this issue:

https://www.labkey.org/issues/home/Developer/issues/details.view?issueId=36231

It sounds like we're both OK with the code from this merge though.